tests/robot_crawler_test.php

```diff
@@ -136,6 +136,14 @@ public function test_should_be_authenticated($expected, $myurl, $testurl) {
         $this->assertEquals((bool)$expected, $this->robot->should_be_authenticated($testurl));
         $this->resetAfterTest(true);
     }
+
+    /**
+     * Tests existence of new plugin parameter 'retentionperiod'
+     */
+    public function test_param_retention_exists() {
+        $param = get_config('local_linkchecker_robot', 'retentionperiod');
```

Comment:
This is wrong, you shouldn't ever be asserting config state. What are you trying to test?

What would be a much better and useful test is to add a bunch of fake records, then set the config for last scrape, run the clean task, and then assert that the old records were removed, and the newer ones were not.